data/src/client.rs

```diff
@@ -689,6 +704,38 @@ impl Client {
                     }
                 }
             }
+            Command::Numeric(RPL_WHOSPCRPL, args) => {
+                let target = args.get(2)?;
+
+                if proto::is_channel(target) {
+                    if let Some(channel) = self.chanmap.get_mut(target) {
+                        let tag = args.get(1)?.parse::<u16>().ok();
+
+                        if let Some(WhoStatus::Requested(_, request_tag)) = channel.last_who {
+                            if request_tag == tag {
+                                channel.last_who = Some(WhoStatus::Receiving);
+                                log::debug!("[{}] {target} - WHO receiving...", self.server);
+                            }
+                        }
+
+                        // We requested, don't save to history
+                        if matches!(channel.last_who, Some(WhoStatus::Receiving)) {
+                            // H = Here, G = gone (away)
+                            let flags = args.get(4)?.chars().collect::<Vec<char>>();
+                            let away = *(flags.first()?) == 'G';
+
+                            let lookup = User::from(Nick::from(args[3].clone()));
+
+                            if let Some(mut user) = channel.users.take(&lookup) {
+                                user.update_away(away);
+                                channel.users.insert(user);
+                            }
```

Comment:
Let's put this logic in the same block as `if request_tag == tag` and only `return None` here. 

This is identical to the logic above in RPL_WHO, maybe we factor this logic into a new function so we don't need to maintain it twice?